crypto/x509/x_all.c

```diff
@@ -74,12 +74,48 @@ int NETSCAPE_SPKI_verify(NETSCAPE_SPKI *a, EVP_PKEY *r)
                             &a->sig_algor, a->signature, a->spkac, r);
 }
 
+/*
+ * Drop empty SKID or AKID X509 extensions that might result from:
+ * "...KeyIdentifier = none"
+ */
+static void drop_empty_cert_keyid_exts(X509 *x)
+{
+    const STACK_OF(X509_EXTENSION) *exts = X509_get0_extensions(x);
+    int i, n;
+
+    if (exts == NULL || (n = sk_X509_EXTENSION_num(exts)) < 1)
+        return;
+
+    for (i = 0; i < n; ++i) {
+        X509_EXTENSION *ext = sk_X509_EXTENSION_value(exts, i);
+        ASN1_STRING *der = X509_EXTENSION_get_data(ext);
+
+        /*
+         * Empty OCTET STRINGs and empty SEQUENCEs encode to just two bytes of
+         * tag (0x04 or 0x30) and length (0x00).
+         */
+        if (der->length == 2 && (der->data[0] == 0x04 || der->data[0] == 0x30)) {
+            ASN1_OBJECT *obj = X509_EXTENSION_get_object(ext);
+            ASN1_OBJECT *skid = OBJ_nid2obj(NID_subject_key_identifier);
+            ASN1_OBJECT *akid = OBJ_nid2obj(NID_authority_key_identifier);
+
+            if (OBJ_cmp(obj, skid) != 0 && OBJ_cmp(obj, akid) != 0)
```

Comment:
```suggestion
            int nid = OBJ_obj2nid(X509_EXTENSION_get_object(ext));

            if (nid != NID_subject_key_identifier && nid != NID_authority_key_identifier)
```

Reply:
Yes, but also here, this would be not only a more efficient, but also 50% more concise.